Assets/Scripts/Framework/Gameplay/PerformanceTracking/PerformanceTracker.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+using ProefExamen.Framework.Gameplay.LaneSystem;
+using ProefExamen.Framework.Gameplay.Values;
+using ProefExamen.Framework.Utils;
+
+namespace ProefExamen.Framework.Gameplay.PerformanceTracking
+{
+    /// <summary>
+    /// A class responsible for tracking the performance of the player.
+    /// </summary>
+    public class PerformanceTracker : AbstractSingleton<PerformanceTracker>
+    {
+        [Header("The current performance")]
+        [SerializeField]
+        private PerformanceResult _newResult;
+
+        /// <summary>
+        /// A list containing HighScores for levels.
+        /// </summary>
+        private List<PerformanceResult> _highscores = null;
+
+        /// <summary>
+        /// An action that shares the performance on a level.
+        /// </summary>
+        public Action<ScoreCompletionStatus> OnScoreCompletion;
+
+        private void Awake() => LoadData();
+
+        private void Start() => LaneManager.Instance.OnNoteHit += RegisterNewHit;
+
+        private void LoadData()
+        {
+            PerformanceTrackerData performanceTrackerData = JsonUtility.FromJson<PerformanceTrackerData>(PlayerPrefs.GetString("highscores"));
```

Comment:
```suggestion
            PerformanceTrackerData performanceTrackerData = 
                 JsonUtility.FromJson<PerformanceTrackerData>(PlayerPrefs.GetString("highscores"));
```
Above 120 characters.